Manage events view

timgl opened the pull request:
## Changes

Moved live events and actions into one view. Also added Event and Property usage and volume, so it becomes a "manage events" view.

Stuff to do next, but wanted to keep this PR contained:
- Rename actions to either "Custom Events" or "Synthetic Events"
- Make all "actions/events" filters into just 'events' filters, and show both actions and events

![image](https://user-images.githubusercontent.com/1727427/98694028-46bcff80-2371-11eb-84f3-6d397c186aa2.png)

## Checklist

- [ ] All querysets/queries filter by Organization, Team, and User (if this PR affects ANY querysets/queries).
- [ ] Django backend tests (if this PR affects the backend).
- [ ] Cypress end-to-end tests (if this PR affects the frontend).

fuziontech committed:
Finish the local dev w/ proto setup

timgl committed:
WIP manage events view

timgl committed:
Add task, add interface etc

timgl committed:
Move everything to 'manage events' view

timgl committed:
Move all settings into single dropdown (can be reverted)

timgl committed:
Urls for tabs

timgl committed:
Merge remote-tracking branch 'origin' into manage-events-view

timgl committed:
Merge remote-tracking branch 'origin' into manage-events-view

yakkomajuri commented:
I pushed back in the past against the name "tagged events", but between that and "custom" or "synthetic", I much prefer "tagged". 

Custom events is not bad, but we already have something with that name. Synthetic is just too long and complex.

yakkomajuri commented on frontend/src/scenes/actions/ActionsTable.js:
```diff
@@ -108,91 +110,66 @@ export function ActionsTable() {
-                                <div className="title">Multiple grouping</div>
-                                <div className="description">Group multiple sets of events into a single action.</div>
-                            </div>
+                    Actions can retroactively group one or more raw events to help provide consistent analytics.{' '}
```
"consistent" is not the best word here - implies that not using them is inconsistent.

yakkomajuri commented on frontend/src/scenes/actions/ActionsTable.js:
```diff
@@ -108,91 +110,66 @@ export function ActionsTable() {
+                    </div>
+                    <div className="t-element">
+                        <div>
+                            <img src={imgStandardized} alt="" />
```
This is probably all over the place, so it's not about this PR necessarily, but we should have alts on images for accessibility and site scores and things

paolodamico replied:
Well I would argue that for this case the text underneath is the alt description, I don't think it makes sense to add extra information. Also, I don't think this will count towards scoring at all as this is behind authentication.

timgl committed:
Fix migration

timgl committed:
Clickhouse and humanize volume

timgl committed:
Fix cypress test

timgl committed:
Fix sidebar cypress

timgl committed:
Fix cypress again

timgl committed:
Fix some small issues

timgl commented:
@paolodamico this is ready for review

paolodamico commented on frontend/src/scenes/events/Events.tsx:
```diff
@@ -0,0 +1,71 @@
+            <PageHeader title="Manage Events" />
+            <Tabs tabPosition="top" animated={false} activeKey={tab} onTabClick={setTab}>
+                <Tabs.TabPane tab="Live Events" key="live">
+                    <i>See all events that are being sent to this team in real time.</i>
```
"sent to this **project**" (same for properties table), "near real-time" maybe?

Also following new design guidelines, this should not be italicized.

paolodamico reviewed (changes requested):
Made a few comments on the code. Here are my comments UX-wise.

1. At first glance it was quite confusing to have "Live Events" & "Events". Maybe change "Events" to something like "Events Volume" or "Manage Events"?

    - Random raw thought, maybe have a "Stats" tab and then have a button group inside to switch between properties & events?

    <img width="428" alt="image" src="https://user-images.githubusercontent.com/5864173/99043472-33cf4880-2586-11eb-8d60-5cc2d1672dd5.png">

1. The title of the page is always "Manage Events" which is confusing when you're checking other stuff (like actions).
    <img width="373" alt="image" src="https://user-images.githubusercontent.com/5864173/99043667-76912080-2586-11eb-88ca-e9b584bcfa89.png">

1. It was a bit unexpected to not see changes in the numbers immediately (I'm particularly concerned that users try to make use of this feature for debugging purposes), and the only mention of a delay in calculation is in a tooltip. I would very clearly add some description at the top of both the event usage & properties usage table that explains that these are intended for statistical purposes and data may have a delay of up to 2 hours or something like that.

1. On the empty state it was a bit confusing to see the event list and everything with 0, perhaps just hide the entire table if all numbers are 0?
     <img width="998" alt="image" src="https://user-images.githubusercontent.com/5864173/99044641-fa97d800-2587-11eb-8fc5-9ccf9c251b84.png">

1. The way 30-day query volume is described I would've understood that this included any query made in insights, but it only counts stuff that's on dashboards. Not entirely sure if the best thing is to count all the insights stuff too or just adjust the description.

1. The properties table is initially shown as sorted alphabetically but it's sorted by custom/PostHog properties first.
    <img width="143" alt="image" src="https://user-images.githubusercontent.com/5864173/99044908-6c702180-2588-11eb-9554-d8225bd6f20d.png">


Finally, not an expert in celery task execution infrastructure, perhaps it'd be a good idea that someone else takes a look to make sure our infrastructure can handle this gracefully?

timgl committed:
Address comments

timgl committed:
Corect naming

timgl committed:
Merge remote-tracking branch 'origin' into manage-events-view

timgl committed:
Fix test'

timgl commented:
@paolodamico Thanks for the review. Have addressed all the individual comments.

1. Done
2. Done
3. Done, I just added `(delayed by up to an hour)` in the column. It's not super pretty but it is clear
4. Hm, not sure i can be any more obvious than the big orange warning sign? It's still useful to have the list of events even if you don't have the volume data
5. It does include all of the insights history? The model name `DashboardItem` is a relic from the past but every time you run a query we create an object on that model
6. Yep, addressed this by putting those props at the bottom

(7) have tested this, the property volumes won't work for our largest client but for everyone else it will

paolodamico reviewed (approved):
Great, lgtm!

Noting 3 very minor things for a future improvement:
- Adding the delay note someplace else that looks better and is a bit more obvious.
- The info icons cause double tooltips to be displayed, perhaps turn off the sorting tooltip or add the information elsewhere.
    <img width="470" alt="image" src="https://user-images.githubusercontent.com/5864173/99072561-32187b80-25ac-11eb-9e97-1968b56907f2.png">
- The table default sorting is great (custom/PostHog properties, alphabetically), the only bug is that by default we show ascending sorting on the first column.
    <img width="277" alt="image" src="https://user-images.githubusercontent.com/5864173/99072679-65f3a100-25ac-11eb-900f-42c0fd045feb.png">
